src/main/java/edu/ucla/library/libcal/services/LibCalProxyServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+
+package edu.ucla.library.libcal.services;
+
+import edu.ucla.library.libcal.Config;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.HttpRequest;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.predicate.ResponsePredicate;
+import io.vertx.ext.web.codec.BodyCodec;
+
+/**
+ * The implementation of LibCalProxyService.
+ */
+public class LibCalProxyServiceImpl implements LibCalProxyService {
+
+    /**
+     * A Vert.x instance.
+     */
+    private final Vertx myVertx;
+
+    /**
+     * App config in Json format.
+     */
+    private final JsonObject myConfig;
+
+    LibCalProxyServiceImpl(final Vertx aVertx, final JsonObject aConfig) {
+        myVertx = aVertx;
+        myConfig = aConfig;
+    }
+
+    @Override
+    public Future<String> getLibCalOutput(final String anOAuthToken, final String aQuery) {
+        /*
+         * for some reason, JsonObject has trouble parsing the output from api/1.1/hours/[id]
```

Comment:
The array that API sends back can't be deserialized to JsonObject, but rather JsonArray. Due to the variation among the available LibCal APIs, I think returning `Future<String>` as you have done is appropriate.

```suggestion
         * for some reason, JsonObject has trouble parsing the output from /api/1.1/hours/[id]
```